packages/flutter/test/material/chip_test.dart

```diff
@@ -6082,6 +6083,60 @@ void main() {
       isNot(paints..rrect(color: hoverColor)..rect(color: themeDataHoverColor)),
     );
   });
+
+  testWidgets('Chip mouse cursor behavior', (WidgetTester tester) async {
+    const SystemMouseCursor customCursor = SystemMouseCursors.grab;
+
+    await tester.pumpWidget(wrapForChip(
+      child: const Center(
+        child: Chip(
+          mouseCursor: customCursor,
+          label: Text('Chip'),
+        ),
+      ),
+    ));
+
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse, pointer: 1);
+    await gesture.addPointer(location: const Offset(10, 10));
+    await tester.pumpAndSettle();
+    expect(
+      RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1),
+      SystemMouseCursors.basic,
+   );
+
+    final Offset chip = tester.getCenter(find.text('Chip'));
+    await gesture.moveTo(chip);
+    await tester.pumpAndSettle();
+    expect(
+      RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1),
+      customCursor,
+    );
+  });
+
+  testWidgets('Mouse cursor resolves in disabled states', (WidgetTester tester) async {
+    tester.binding.focusManager.highlightStrategy = FocusHighlightStrategy.alwaysTraditional;
+    final FocusNode focusNode = FocusNode(debugLabel: 'Chip');
+    addTearDown(focusNode.dispose);
+
+    // Test unfocused case.
+    await tester.pumpWidget(
+      wrapForChip(
+        child: Center(
+          child: Chip(
+            mouseCursor: const _ChipMouseCursor(),
```

Comment:
```suggestion
            mouseCursor: const WidgetStateMouseCursor.fromMap(
              <WidgetStatesConstraint, MouseCursor>{
                WidgetState.disabled: SystemMouseCursors.forbidden,
              },
            ),
```

The `_ChipMouseColor` class is probably not needed (Same for the other test file). While it works, I think it makes the test more difficult to understand. Passing directly a `WidgetStateMouseCursor` created from the `fromMap` factory is also what most users will probably do.
Did I miss something?

Reply:
It's way better! I've implemented the changes in [Use WidgetStateMouseCursor.fromMap](https://github.com/flutter/flutter/pull/159422/commits/1bb971b53fca6c732abacb82a0ac42a9d290fde3)